lib/initializer.rb

```diff
@@ -0,0 +1,56 @@
+require_relative './game.rb'
+require_relative './board/board.rb'
+require_relative './board/board_checker.rb'
+require_relative './players/ai.rb'
+require_relative './players/human.rb'
+require_relative './players/player.rb'
+require_relative './ui/user_interface.rb'
+
+class Initializer
+  attr_accessor :logic1
+
+  def initialize(size, empty_char)
+    @size = size
+    @empty_char = empty_char
+    @player1
+    @player2
+  end
+
+  def start
+    board = Board.new(@size, @empty_char)
+    checker = BoardChecker.new(@size, @empty_char)
+    user_interface = UserInterface.new(board, 'X', 'O')
+    ai_logic = AI.new(checker, 'O', 'X')
+    human_logic = Human.new(board, user_interface, 'X')
+
+    welcome(user_interface)
+    selection = who_goes_first?(user_interface)
+    setup_players(selection, ai_logic, human_logic, user_interface)
+
+    game = Game.new(board, checker, @player1, @player2)
+    play(game)
```

Comment:
Oh, ok cool I see what's going on and why you set the players to null.

I mentioned in our IPM that I'm not a big fan of initializing objects in constructors, but now that I'm looking this over, I think a better way to phrase my opinion is that I _am_ a big fan of injecting dependencies. I find that it makes testing, managing, and extending an application easier. Try to avoid initializing anything inside of another class.

There's always going to be some type of `main` method that starts an app. I find that it's a good place to procedurally wire up all of your dependencies.